File: Front-End/src/services/article-category.service.ts
import axiosClient from "@/configurations/axios.config";
import type {
  CreateArticleCategoryRequest,
  ArticleCategoryResponse,
  ArticleCategoryListResponse,
} from "@/types/article-category.type";

export const articleCategoryService = {
  getCategories: async (page = 1, limit = 7, title = "") => {
    const response = await axiosClient.get<ArticleCategoryListResponse>(
      `/article-categories?page=${page}&limit=${limit}&title=${encodeURIComponent(title)}`
    );
    return response.data;
  },

  getCategoryById: async (id: number) => {
    const response = await axiosClient.get<ArticleCategoryResponse>(
      `/article-categories/${id}`
    );
    return response.data;
  },

  createCategory: async (data: CreateArticleCategoryRequest) => {
    const response = await axiosClient.post<ArticleCategoryResponse>(
      `/article-categories`,
      data
    );
    return response.data;
  },

  updateCategory: async (id: number, data: CreateArticleCategoryRequest) => {
    const response = await axiosClient.put<ArticleCategoryResponse>(
      `/article-categories/${id}`,
      data
    );
    return response.data;
  },


  getCategoryBySlug: async (slug: string) => {
    const response = await axiosClient.get<ArticleCategoryResponse>(
      `/article-categories/slug/${slug}`
    );
    return response.data;
  },

  deleteCategory: async (id: number) => {
    await axiosClient.delete(`/article-categories/${id}`);
  },
};

File: Back-End/src/main/java/iuh/fit/ecommerce/services/ArticleCategoryService.java
package iuh.fit.ecommerce.services;

import iuh.fit.ecommerce.dtos.request.article.ArticleCategoryAddRequest;
import iuh.fit.ecommerce.dtos.response.article.ArticleCategoryResponse;
import iuh.fit.ecommerce.dtos.response.base.ResponseWithPagination;
import java.util.List;

public interface ArticleCategoryService {

    ArticleCategoryResponse createCategory(ArticleCategoryAddRequest request);

    ArticleCategoryResponse getCategoryBySlug(String slug);

    ArticleCategoryResponse getCategoryById(Long id);

    ArticleCategoryResponse updateCategory(Long id, ArticleCategoryAddRequest request);

    void deleteCategory(Long id);

    ResponseWithPagination<List<ArticleCategoryResponse>> getAllCategories(int page, int limit, String title);
}